### DISCUSSION TITLE
Is there a way to get currently opened tab URL from CLI?

### DISCUSSION THREAD

**Discussion by sprnza** [qutebrowser/qutebrowser#7657] [General] [answered]
I'd like to get an URL of currently opened tab from command line. Something like
```
> qutebrowser ':get_url'
> youtube.com
```
I get it ising sqlite3 from the qutebrowser database wich is unreliable because data is sorted by opened time not access time. So if I open youtube.com command
```
sqlite3 history.sqlite 'select title from History order by atime desc limit 1;
```
returns Youtube. But if I open another tab (google.com) and then return back to Youtube I get title of the last opened tab title (Google).

**Comment by mschilli87** [comment]
Which URL do you expect if you have 4 qutebrowser windows from 3 sessions running, one of them with a different basedir and all of them with various tabs when your terminal has focus?

**Comment by sprnza** [comment]
yeah, haven'  considered this. Sounds like an impossible task

**Comment by neeasade** [comment, marked as answer]
I do this by dumping the current session, and selecting the first window with yq (I only have one qb window open ever though): https://github.com/neeasade/dotfiles/blob/a44de95b687f751401ef2adb38678535214e1e65/bin/bin/qb_active_url#L5

session dump script is: https://github.com/neeasade/dotfiles/blob/a44de95b687f751401ef2adb38678535214e1e65/bin/bin/qb_session_dump#L12

one quirk of this is the auto-session is not updated when deleting a tab. to fix this, I save the session on `d`: https://github.com/neeasade/dotfiles/blob/a44de95b687f751401ef2adb38678535214e1e65/net/.config/qutebrowser/config.py#L82C1-L89

**Reply by sprnza** [reply]
Wow! Thank you!
What package provides `bb` used [here](https://github.com/neeasade/dotfiles/blob/a44de95b687f751401ef2adb38678535214e1e65/bin/bin/qb_session_dump#L12)? Archlinux AUR contains only this bb which I don't think is the right choice ))
```
aur/bb 1.3rc1-15 (+24 1.32)
    ASCII-art demo that uses AAlib
```

**Reply by sprnza** [reply]
Never mind! I've found it (babashka). Thank you for sharing!
